tests/tools/test_openroad.py

```
@@ -1718,6 +1718,60 @@ def test_openroad_cleanup_synth_parameter_remove_dead_logic():
+    task = synth_cleanup.CleanupSynthTask.find_task(asic_gcd)
+    task.set_openroad_removebuffers(remove)
+    task.set_openroad_repairsynthtiming(repair)
+
+    node = SchedulerNode(asic_gcd, "cleanup.clean", "0")
+    with node.runtime():
+        if legal:
+            assert node.setup() is True
+        else:
+            with pytest.raises(ValueError,
+                               match="^remove_synth_buffers and repair_synth_timing are "
+                                     "alternatives, not additive: .* Enable one or neither\\.$"):
```

Ruff (0.16.1)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)